presto-tpcds/src/main/java/com/facebook/presto/tpcds/TpcdsRecordSet.java

```diff
@@ -0,0 +1,191 @@
+
+import static com.facebook.presto.spi.type.Decimals.rescale;
+import static com.facebook.presto.tpcds.TpcdsMetadata.getPrestoType;
+import static com.google.common.base.Preconditions.checkState;
+import static java.lang.Double.parseDouble;
+import static java.lang.Integer.parseInt;
+import static java.lang.Long.parseLong;
+import static java.util.Objects.requireNonNull;
+
+public class TpcdsRecordSet
+        implements RecordSet
+{
+    private final Results results;
+    private final List<Column> columns;
+
+    private final List<Type> columnTypes;
+
+    public TpcdsRecordSet(Results results, List<Column> columns)
+    {
+        requireNonNull(results, "results is null");
+
+        this.results = results;
+        this.columns = ImmutableList.copyOf(columns);
+        ImmutableList.Builder<Type> columnTypes = ImmutableList.builder();
+        for (Column column : columns) {
+            columnTypes.add(getPrestoType(column.getType()));
+        }
+        this.columnTypes = columnTypes.build();
+    }
+
+    @Override
+    public List<Type> getColumnTypes()
+    {
+        return columnTypes;
+    }
+
+    @Override
+    public RecordCursor cursor()
+    {
+        return new TpcdsRecordCursor(results.iterator(), columns);
+    }
+
+    public class TpcdsRecordCursor
+            implements RecordCursor
+    {
+        private final Iterator<List<List<String>>> rows;
+        private final List<Column> columns;
+        private List<String> row;
+        private boolean closed;
+
+        public TpcdsRecordCursor(Iterator<List<List<String>>> rows, List<Column> columns)
+        {
+            this.rows = requireNonNull(rows);
+            this.columns = requireNonNull(columns);
+        }
+
+        @Override
+        public long getTotalBytes()
+        {
+            return 0;
+        }
+
+        @Override
+        public long getCompletedBytes()
+        {
+            return 0;
+        }
+
+        @Override
+        public long getReadTimeNanos()
+        {
+            return 0;
+        }
+
+        @Override
+        public Type getType(int field)
+        {
+            return getPrestoType(columns.get(field).getType());
+        }
+
+        @Override
+        public boolean advanceNextPosition()
+        {
+            if (closed || !rows.hasNext()) {
+                closed = true;
+                row = null;
+                return false;
+            }
+
+            row = rows.next().get(0);
+            return true;
+        }
+
+        @Override
+        public boolean getBoolean(int field)
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public long getLong(int field)
+        {
+            checkState(row != null, "No current row");
+            Column column = columns.get(field);
+            if (column.getType().getBase() == ColumnType.Base.DATE) {
+                return LocalDate.parse(row.get(column.getPosition())).toEpochDay();
```

Comment:
`LocalDate` and `LocalTime` are much less efficient than Joda because they create an object with fields and then convert to long where as Joda converts to a long directly.